azure/services/aso/tags.go

```diff
@@ -49,17 +46,7 @@ func reconcileTags[T genruntime.MetaObject](t TagsGetterSetter[T], existing T, r
 			}
 		}
 
-		existingTags = t.GetActualTags(existing)
-		// Wait for tags to converge so we know for sure which ones are deleted from additionalTags (and
-		// should be deleted) and which were added manually (and should be kept).
-		if !reflect.DeepEqual(t.GetDesiredTags(existing), existingTags) &&
-			existing.GetAnnotations()[asoannotations.ReconcilePolicy] == string(asoannotations.ReconcilePolicyManage) {
```

Comment:
> No, because the tags in the status are no longer used to calculate the new spec tags.

Ok, so with this code block dropped, CAPZ is no longer going to wait for `spec.tags` to show up in `status.tags`.

Earlier, we were waiting for `spec.tags` to reflect in -> `status.tags `(reference conversation: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4149/files#r1362470705)

Just curious, do you remember the reason to consider the earlier path of _"wait for tags to show up in status"_ instead of using `spec.tags` to begin with? I hope we are not missing anything, this implementation looks good to me overall.

Reply:
> Just curious, do you remember the reason to consider the earlier path of _"wait for tags to show up in status"_ instead of using `spec.tags` to begin with?

I didn't consider this then, but I think I would've preferred this way then too.